main.go

```diff
@@ -209,11 +210,44 @@ func main() {
 
 		if cfg.url != nil {
 			ctx, cancel := context.WithCancel(ctx)
+			// Create Reverse Proxy.
 			p := httputil.ReverseProxy{
 				Director: func(request *http.Request) {
 					request.URL.Scheme = cfg.url.Scheme
+					// Set the Host at both request and request.URL objects.
 					request.Host = cfg.url.Host
 					request.URL.Host = cfg.url.Host
+					// Derive path from the paths of configured URL and request URL.
+					request.URL.Path, request.URL.RawPath =
+						func(a, b *url.URL) (path, rawpath string) {
```

Comment:
can we extract this anonymous function into its own func? I think this would help with readability